pkg/query/storeset.go

```diff
@@ -197,20 +231,64 @@ func (s *StoreSet) Update(ctx context.Context) {
 	s.mtx.Lock()
 	defer s.mtx.Unlock()
 
+	//Also record the number of occurrences of external label combinations.
+	externalLabelStores := map[string][]*storeRef{}
+	for _, st := range s.stores {
+		externalLabels := externalLabelsFromStore(st)
+		externalLabelStores[externalLabels] = append(externalLabelStores[externalLabels], st)
+	}
+
 	// Remove stores that where not updated in this update, because were not in s.peerAddr() this time.
 	for addr, st := range s.stores {
 		if _, ok := stores[addr]; ok {
-			continue
+			externalLabelOccurrences := len(externalLabelStores[externalLabelsFromStore(st)])
+			if externalLabelOccurrences == 1 {
+				continue
+			} else {
+				level.Warn(s.logger).Log("msg", fmt.Sprintf("dropping store with address %s, external labels are not unique", st.addr))
```

Comment:
Can we keep structured logs and do `("msg", "dropping store, external labels are not unique", "addr", st.addr)` instead?